modules/misunderstood/src/backend/db.ts

```
@@ -103,6 +106,75 @@ export default class Db {
     }))
   }
 
+  async importData(botId: string, events: FlaggedEvent[]): Promise<void> {
+    const botGhost = this.ghostForBot(botId)
+
+    // Only deal with events that have a resolution defined. They can be null and pass the type check
+    // because the type allows for un-resolved events, but we can't import those here
+    events = events.filter(d => d.resolution && d.resolutionType)
+    if (events.length === 0) {
+      return
+    }
+
+    // sort intents and QnAs
+    const qnaEvents = events.filter(({ resolutionType }) => resolutionType === RESOLUTION_TYPE.qna)
+    const nluEvents = events.filter(({ resolutionType }) => resolutionType === RESOLUTION_TYPE.intent)
+
+    // Assert the intents and QnAs already exist
+    const fileList = [
+      ...(await botGhost.directoryListing('intents', '*.json')),
+      ...(await botGhost.directoryListing('qna', '*.json'))
+    ]
+    let missing: Array<string> = []
+    for (const event of [...qnaEvents, ...nluEvents]) {
+      if (!fileList.includes(`${event.resolution}.json`)) {
+        missing.push(event.resolution)
+      }
+    }
+    if (missing.length) {
```

raise with JF - should we raise or create missing